api/src/main/java/org/openmrs/Allergy.java

```diff
@@ -15,26 +15,60 @@
 import java.util.List;
 import java.util.UUID;
 
+import javax.persistence.AttributeOverride;
+import javax.persistence.AttributeOverrides;
+import javax.persistence.CascadeType;
+import javax.persistence.Column;
+import javax.persistence.Embedded;
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.Id;
+import javax.persistence.JoinColumn;
+import javax.persistence.ManyToOne;
+import javax.persistence.OneToMany;
+import javax.persistence.OrderBy;
+import javax.persistence.Table;
+
 import org.apache.commons.lang3.StringUtils;
+import org.hibernate.annotations.LazyCollection;
+import org.hibernate.annotations.LazyCollectionOption;
 import org.openmrs.util.OpenmrsUtil;
 
 /**
  * Represent allergy
  */
+@Entity
+@Table(name = "allergy")
 public class Allergy extends BaseChangeableOpenmrsData {
 	
 	public static final long serialVersionUID = 1;
 	
+	@Id
+	@GeneratedValue
+	@Column(name = "allergy_id")
 	private Integer allergyId;
 	
+	@ManyToOne(optional = false)
+	@JoinColumn(name = "patient_id")
 	private Patient patient;
 	
+	@Embedded
+	@AttributeOverrides({ @AttributeOverride(name = "allergenType", column = @Column(name = "allergen_type")),
+	        @AttributeOverride(name = "codedAllergen", column = @Column(name = "coded_allergen")),
+	        @AttributeOverride(name = "nonCodedAllergen", column = @Column(name = "non_coded_allergen")) })
 	private Allergen allergen;
 	
+	@ManyToOne
+	@JoinColumn(name = "severity_concept_id")
 	private Concept severity;
 	
+	@Column(name = "comments", length = 1024)
 	private String comments;
 	
+
+	@OneToMany(mappedBy = "allergy", cascade = CascadeType.ALL, orphanRemoval = true)
+	@LazyCollection(LazyCollectionOption.FALSE)
+	@OrderBy("allergy_id")
```

Comment:
There was no ordering set in the hbm file

Reply:
And I think by default they are ordered by allergy_id anyways

Reply:
Did you see this comment?

Reply:
yes @wluyima  i have seen the comment n fixed...